ophyd/signal.py

```diff
@@ -457,6 +481,30 @@ def read(self):
         value = self.get()
         return {self.name: {"value": value, "timestamp": self.timestamp}}
 
+    def _infer_value_kind(
+        self, kind, inference_funcs={"dtype": data_type, "shape": data_shape}
```

Comment:
rather than a string + a mutable default + a dict lookup would could we just pass the inference function?

Reply:
😆 sure ; done in latest version